### PR TITLE
chore: remove deepsource for good

### PR DESCRIPTION
**Pull request by vandot**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2486)

### CODE REVIEW

**Review by mrekucci — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2486)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @acud and @istae)

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -49,6 +49,9 @@ jobs:
     - name: Vet
       if: matrix.os == 'ubuntu-latest'
       run: make vet
+    - name: Coverage
+      if: matrix.os == 'ubuntu-latest'
+      run: go test -coverprofile=cover.out ./...
```
Comment (acud) [inline .github/workflows/go.yml:54]: i think this step is not needed. you can just add `-coverprofile=cover.out` to the test step below (Test with Race detector). the coveralls only run on the ubuntu build (the one that uses the test with race detector). not sure if `make` will like the extra params though, we might need to change the makefile

Comment (vandot) [reply .github/workflows/go.yml:54]: @acud I thought that but `make` doesn't accept additional argument (unless I do some changes) and we will create `cover.out` on mac as well (which is not a real issue), if we do it like this tests will be cached and we will not slow down whole pipeline

Comment (acud) [reply .github/workflows/go.yml:54]: cool :+1:

**Review by acud — APPROVED** [review]

**Pull request merged**